diff -ur linux-rpi-3.6.y.orig/arch/arm/mach-bcm2708/bcm2708.c linux-rpi-3.6.y/arch/arm/mach-bcm2708/bcm2708.c
--- linux-rpi-3.6.y.orig/arch/arm/mach-bcm2708/bcm2708.c	2012-11-28 17:49:53.000000000 +0100
+++ linux-rpi-3.6.y/arch/arm/mach-bcm2708/bcm2708.c	2012-11-29 18:15:43.742624990 +0100
@@ -770,7 +770,7 @@
 	case CLOCK_EVT_MODE_UNUSED:
 	case CLOCK_EVT_MODE_SHUTDOWN:
 	default:
-		printk(KERN_ERR "timer_set_mode: unhandled mode:%d\n",
+		printk("timer_set_mode: unhandled mode:%d\n",
 		       (int)mode);
 		break;
 	}
diff -ur linux-rpi-3.6.y.orig/arch/arm/mach-bcm2708/bcm2708_gpio.c linux-rpi-3.6.y/arch/arm/mach-bcm2708/bcm2708_gpio.c
--- linux-rpi-3.6.y.orig/arch/arm/mach-bcm2708/bcm2708_gpio.c	2012-11-28 17:49:53.000000000 +0100
+++ linux-rpi-3.6.y/arch/arm/mach-bcm2708/bcm2708_gpio.c	2012-11-29 18:16:05.862625551 +0100
@@ -263,7 +263,7 @@
 	struct resource *res;
 	int err = 0;
 
-	printk(KERN_ERR DRIVER_NAME ": bcm2708_gpio_probe %p\n", dev);
+	printk(DRIVER_NAME ": bcm2708_gpio_probe %p\n", dev);
 
 	ucb = kzalloc(sizeof(*ucb), GFP_KERNEL);
 	if (NULL == ucb) {
